test/data/EnumerableSet.ts

```diff
@@ -0,0 +1,195 @@
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { bnToBytes32 } from '@solidstate/library';
+import {
+  EnumerableSetBytes32Mock,
+  EnumerableSetBytes32Mock__factory,
+} from '@solidstate/typechain-types';
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+
+describe('EnumerableSet', async () => {
+  describe('Bytes32Set', async () => {
+    let instance: EnumerableSetBytes32Mock;
+    let deployer: SignerWithAddress;
+
+    beforeEach(async () => {
+      [deployer] = await ethers.getSigners();
+      instance = await new EnumerableSetBytes32Mock__factory(deployer).deploy();
+    });
+
+    describe('__internal', () => {
+      const zeroBytes32 = bnToBytes32(ethers.constants.Zero);
+      const oneBytes32 = bnToBytes32(ethers.constants.One);
+      const twoBytes32 = bnToBytes32(ethers.constants.Two);
+
+      describe('#at(uint256)', () => {
+        it('returns the value corresponding to index provided', async () => {
+          await instance['add(bytes32)'](zeroBytes32);
+          await instance['add(bytes32)'](oneBytes32);
+          await instance['add(bytes32)'](twoBytes32);
+
+          expect(await instance['at(uint256)'](0)).to.equal(zeroBytes32);
+          expect(await instance['at(uint256)'](1)).to.equal(oneBytes32);
+          expect(await instance['at(uint256)'](2)).to.equal(twoBytes32);
+        });
+
+        describe('reverts if', function () {
+          it('index out of bounds', async () => {
+            await expect(instance['at(uint256)'](0)).to.be.revertedWith(
```

Comment:
Perhaps wait until #139 is merged before continuing with tests?

Reply:
Actually, this is minor and can be retroactively fixed.

Reply:
Merged in that branch, can be fixed now.